r/src/recordbatch.cpp

```diff
@@ -246,6 +246,7 @@ std::shared_ptr<arrow::RecordBatch> RecordBatch__from_arrays__known_schema(
   SEXP names = Rf_getAttrib(lst, R_NamesSymbol);
 
   auto fill_array = [&arrays, &schema](int j, SEXP x, SEXP name) {
+    name = Rf_mkCharCE(Rf_translateCharUTF8(name), CE_UTF8);
```

Comment:
There are a few places where I `Rf_mkCharCE()` and then immediately call `CHAR()`, which IIUC is boxing in a SEXP and then immediately unboxing it. Maybe that can be eliminated some places or done better?

Reply:
I don't think R offers api for this